lntest/itest/lnd_onchain_test.go

```diff
@@ -379,3 +382,237 @@ func testAnchorReservedValue(net *lntest.NetworkHarness, t *harnessTest) {
 		t.Fatalf("expected 1 output instead have %v", len(sweepTx.TxOut))
 	}
 }
+
+// genAnchorSweep generates a "3rd party" anchor sweeping from an existing one.
+// In practice, we just re-use the existing witness, and track on our own
+// output producing a 1-in-1-out transaction.
+func genAnchorSweep(t *harnessTest, net *lntest.NetworkHarness,
+	aliceAnchor *sweptOutput, anchorCsv uint32) *btcutil.Tx {
+
+	// At this point, we have the transaction that Alice used to try to
+	// sweep her anchor. As this is actually just something anyone can
+	// spend, just need to find the input spending the anchor output, then
+	// we can swap the output address.
+	aliceAnchorTxIn := func() wire.TxIn {
+		sweepCopy := aliceAnchor.SweepTx.Copy()
+		for _, txIn := range sweepCopy.TxIn {
+			if txIn.PreviousOutPoint == aliceAnchor.OutPoint {
+				return *txIn
+			}
+		}
+
+		t.Fatalf("anchor op not found")
+		return wire.TxIn{}
+	}()
+
+	// We'll set the signature on the input to nil, and then set the
+	// sequence to 16 (the anchor CSV period).
+	aliceAnchorTxIn.Witness[0] = nil
+	aliceAnchorTxIn.Sequence = anchorCsv
+
+	minerAddr, err := net.Miner.NewAddress()
+	if err != nil {
+		t.Fatalf("unable to get miner addr: %v", err)
+	}
+	addrScript, err := txscript.PayToAddrScript(minerAddr)
+	if err != nil {
+		t.Fatalf("unable to gen addr script: %v", err)
+	}
+
+	// Now that we have the txIn, we can just make a new transaction that
+	// uses a different script for the output.
+	const anchorSize = 330
+	tx := wire.NewMsgTx(2)
+	tx.AddTxIn(&aliceAnchorTxIn)
+	tx.AddTxOut(&wire.TxOut{
+		PkScript: addrScript,
+		Value:    anchorSize - 1,
```

Comment:
Curious, why minus 1?

Reply:
Just to pay 1 sat fee to the miner. It can likely work w/o that, but then we'd possibly make the tests less deterministic if the miner had already exhaust its "penny flood" allotment.

Reply:
By penny flood you mean the free block space provided by the miner right?

Reply:
It's a sort of bandwidth allotment for transaction belwo the min relay fee: https://github.com/btcsuite/btcd/blob/97732e52810c99ba789f623c8c2ccb8290fbadf1/mempool/mempool.go#L1173

I think these days bitcoind does somethign slightly different, but it's a defense against ppl just relaying  abunch of zero fee transactions across the network.